packages/langchain/src/orchestration/client.ts

```diff
@@ -108,14 +112,86 @@ export class OrchestrationClient extends BaseChatModel<
 
     const content = res.getContent();
 
-    // TODO: Add streaming as soon as we support it
     await runManager?.handleLLMNewToken(
       typeof content === 'string' ? content : ''
     );
 
     return mapOutputToChatResult(res.data);
   }
 
+  /**
+   * Stream response chunks from the Orchestration client.
+   * @param messages - The messages to send to the model.
+   * @param options - The call options.
+   * @param runManager - The callback manager for the run.
+   * @returns An async generator of chat generation chunks.
+   */
+  override async *_streamResponseChunks(
+    messages: BaseMessage[],
+    options: typeof this.ParsedCallOptions,
+    runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<ChatGenerationChunk> {
+    const controller = new AbortController();
+    if (options.signal) {
+      options.signal.addEventListener('abort', () => controller.abort());
+    }
+
+    const orchestrationMessages =
+      mapLangChainMessagesToOrchestrationMessages(messages);
+
+    const { inputParams, customRequestConfig } = options;
+    const mergedOrchestrationConfig = this.mergeOrchestrationConfig(options);
+
+    const orchestrationClient = new OrchestrationClientBase(
+      mergedOrchestrationConfig,
+      this.deploymentConfig,
+      this.destination
+    );
+
+    const response = await this.caller.call(() =>
+      orchestrationClient.stream(
+        // TODO: Adapt messagesHistory during https://github.com/SAP/ai-sdk-js-backlog/issues/317
+        { messagesHistory: orchestrationMessages, inputParams },
+        controller,
+        options.streamOptions,
+        customRequestConfig
+      )
+    );
+
+    for await (const chunk of response.stream) {
+      const messageChunk = mapOrchestrationChunkToLangChainMessageChunk(chunk);
+      const tokenIndices = computeTokenIndices(chunk);
+      const finishReason = response.getFinishReason();
+      const tokenUsage = response.getTokenUsage();
+
+      setFinishReason(messageChunk, finishReason);
+      setTokenUsage(messageChunk, tokenUsage);
+      const content = chunk.getDeltaContent() ?? '';
+
+      const generationChunk = new ChatGenerationChunk({
+        message: messageChunk,
+        text: content,
+        generationInfo: { ...tokenIndices }
+      });
+
+      // Notify the run manager about the new token, some parameters are undefined as they are implicitly read from the context.
```

Comment:
[req] Can we document such stuff for future? I wouldn't know why we set some values as undefined, unless I read the langchain code.

Reply:
I would put it as a comment right above this line. Doesn't need to be user-facing, but for us to maintain in the future.